src/distributed_join.cuh

```diff
@@ -370,11 +419,15 @@ distributed_inner_join(
 
     vector<vector<vector<void *> > > left_buckets(over_decom_factor);  // [ibatch, icol, ibucket]
     vector<vector<vector<void *> > > right_buckets(over_decom_factor);  // [ibatch, icol, ibucket]
-    vector<vector<vector<int64_t> > > left_counts(over_decom_factor);  // [ibatch, icol, ibucket]
-    vector<vector<vector<int64_t> > > right_counts(over_decom_factor);  // [ibatch, icol, ibucket]
-    vector<bool> flags(over_decom_factor, false);  // whether each batch has finished communication
+    vector<vector<int64_t> > left_counts(over_decom_factor);  // [ibatch, ibucket]
+    vector<vector<int64_t> > right_counts(over_decom_factor);  // [ibatch, ibucket]
+    vector<std::atomic<bool> > flags(over_decom_factor);  // whether each batch has finished communication
```

Comment:
Clarify why using `std::atomic` here, e.g. that we may have concurrent stores to vector's elements.

Reply:
There's no concurrent stores. `flags` will be written in one thread and read from another. From cpp reference
> When an evaluation of an expression writes to a memory location and another evaluation reads or modifies the same memory location, the expressions are said to conflict. A program that has two conflicting evaluations has a data race

I will add a note about the motivation.